### ISSUE TITLE
Normalize/merge the hotwords

### ISSUE THREAD

**Issue by vbouchet31** [Kitt-AI/snowboy#357]
I don't know if it is applicable to French only but I am wondering if there is any reason to not normalize the hotwords. What I mean by normalize is to use lowercase only for example.

If I check the most popular French keywords, I can see "jarvis", "Jarvis", "JARVIS" or "Bonjour", "bonjour".

I am wondering if some of these hotwords can be merged. I don't think it will increase a lot the number of recordings because I suspect some people recorded multiple versions but that would be less confusing and may help to reach the target to train a public model.

**Comment by chenguoguo** [comment]
Good point. We thought about doing it, but `normalize` is actually a little bit tricky. Case normalization is one of the ways to do the normalization, and works for languages such as English and French, but this doesn't naturally apply to other languages. We ended up leaving it as is. We'll think about it again if we need to do any normalization.

**Comment by vbouchet31** [comment]
Maybe a first approach would be to allow users to merge hotwords or at least to suggest / vote for merges. After X votes, the 2 hotwords are automatically merged. It is not ideal but probably much much easier than automatic "normalization".

**Comment by wibimaster** [comment]
Pleeeease, at least for some cases ; french "Jarvis", jarvis" and "JARVIS" totally count 1919 voices, shoudl be a real big step to the first universal french hotword !

:pray:

**Comment by wibimaster** [comment]
@chenguoguo Little bump, can you tell us if something prevents you from solving the problem for the most popular of French words, Jarvis?
As an exception :)

Thanks !
